src/Sly/NotificationPusher/Adapter/Gcm.php

```diff
@@ -71,21 +71,19 @@ public function push(PushInterface $push)
                 $response = $client->send($message);
                 $responseResults = $response->getResults();
 
-                if ((bool) $response->getSuccessCount()) {
-                    foreach ($tokensRange as $token) {
-                        $device = $push->getDevices()->get($token);
-                        
-                        // map the overall response object
-                        // into a per device response
-                        $tokenResponse = @$responseResults[$token] ?: [];
-                        $tokenResponse = array_merge(
-                            $tokenResponse, 
-                            array_diff_key($response->getResponse(), ['results' => true])
-                        );
-                        $push->addResponse($device, $tokenResponse);
-                        
-                        $pushedDevices->add($device);
-                    }
+                foreach ($tokensRange as $token) {
+                    $device = $push->getDevices()->get($token);
+                    
+                    // map the overall response object
+                    // into a per device response
+                    $tokenResponse = @$responseResults[$token] ?: [];
+                    $tokenResponse = array_merge(
+                        $tokenResponse, 
+                        array_diff_key($response->getResponse(), ['results' => true])
```

Comment:
it seems that it fails because the `getResponse` from the mocked client does not return an array.
But as of zendframework\zendservice-google-gcm\library\ZendService\Google\Gcm\Client.php the response is always given, otherwise an exception is thrown.